src/python/pants/goal/stats_aggregator.py

```diff
@@ -80,6 +104,15 @@ def __call__(
         if not finished:
             return
 
+        output_lines = []
+        if self.output_file:
+            timestamp = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+            # have an empty line between stats of different Pants invocations
+            space = "\n\n" if Path(self.output_file).exists() else ""
+            output_lines.append(
+                f"{space}{timestamp} Executing goals: {','.join(context._run_tracker.goals)}"
```

Comment:
I wonder if there's a better way to obtain the executed goals than `context._run_tracker.goals`.

Reply:
It looks like the run tracker is exposed "publicly" via the `run_tracker` property, so maybe this is the best way:

```suggestion
                f"{space}{timestamp} Executing goals: {','.join(context.run_tracker.goals)}"
```

That said, it looks like there's also some "run info" that particularly has the exact command line. What do you think about using that directly?

```suggestion
                f"{space}{timestamp} Executing: {context.run_tracker.run_information.get("cmd_line")}"
```

Reply:
Oh that's great, many thanks.